app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitOrderExtraQuestionTypeApiController.php

```diff
@@ -444,4 +455,96 @@ public function deleteSubQuestionRule($summit_id, $question_id, $rule_id)
             return $this->deleted();
         });
     }
+
+    /**
+     * @param $summit_id
+     * @return mixed
+     */
+    public function getOwnAttendeeAllowedExtraQuestions($summit_id)
+    {
+        return $this->processRequest(function () use ($summit_id) {
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->getResourceServerContext())->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $type = CheckAttendeeStrategyFactory::Me;
+            $attendee = CheckAttendeeStrategyFactory::build($type, $this->resource_server_context)->check('me', $summit);
+            if (is_null($attendee)) return $this->error404();
+
+            return $this->getAttendeeExtraQuestions($summit_id, $attendee->getId());
+        });
+    }
+
+    /**
+     * @param $summit_id
+     * @param $attendee_id
+     * @return mixed
+     */
+    public function getAttendeeExtraQuestions($summit_id, $attendee_id)
+    {
+        $summit = SummitFinderStrategyFactory::build($this->getSummitRepository(), $this->getResourceServerContext())->find($summit_id);
+        if (is_null($summit)) return $this->error404();
+
+        $attendee = $summit->getAttendeeById(intval($attendee_id));
+        if (is_null($attendee)) return $this->error404();
+
+        return $this->_getAll(
+            function () {
+                return [
+                    'name'      => ['=@', '=='],
+                    'type'      => ['=@', '=='],
+                    'label'     => ['=@', '=='],
+                    'printable' => ['=='],
+                    'usage'     => ['=@', '=='],
+                    'summit_id' => ['=='],
+                    'has_badge_feature_types' => ['=='],
+                    'has_ticket_types' => ['=='],
+                ];
+            },
+            function () {
+                return [
+                    'name'      => 'sometimes|string',
+                    'type'      => 'sometimes|string',
+                    'label'     => 'sometimes|string',
+                    'printable' => 'sometimes|string|in:true,false',
+                    'usage'     => 'sometimes|string',
+                    'summit_id' => 'sometimes|integer',
+                    'has_badge_feature_types' => 'sometimes|string|in:true,false',
+                    'has_ticket_types' => 'sometimes|string|in:true,false',
+                ];
+            },
+            function () {
+                return [
+                    'id',
+                    'name',
+                    'label',
+                    'order',
+                ];
+            },
+            function ($filter) use ($summit, $attendee) {
+                if ($filter instanceof Filter) {
+                    $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit->getId()));
```

Comment:
lets review the filtering condition here
it should be something like
(has_badge_feature_types == false and has_ticket_types == false) or ( allowed_badge_feature_type_id in ( attendees features) OR allowed_ticket_type_id in ( attendees tickets types)

Reply:
also its missing these filters
class==MainQuestion and usage==Ticket

Reply:
done